pkg/minikube/mount/cifs_windows.go

```diff
@@ -0,0 +1,159 @@
+	if err != nil {
+		return errors.Wrap(err,"current user not administrator")
+	}
+
+	if err := hyperv.PowershellCmd("SmbShare\\Remove-SmbShare", "-Name", m.HostPath, "-Force"); err != nil {
+		return errors.Wrap(err,"remove smb share")
+	}
+	return nil
+}
+
+func (m *Cifs) Mount(r mountRunner) error {
+	hostname, err := os.Hostname()
+	if err != nil {
+		return errors.Wrap(err, "get hostname")
+	}
+	user, err := hyperv.CurrentWindowsUser()
+	if err != nil {
+		return errors.Wrap(err,"get current user")
+	}
+
+	// We need the Domain Name of the current user and the User Name. This is used while connecting to the Share from Linux.
+	stringSplitResult := strings.Split(user,"\\")
+	domain := stringSplitResult[0]
+	username := stringSplitResult[1]
+
+	// Only ask the user for the password if it is not found in the Windows Credential Store
+	mountPassword,err := getMountPassword()
+	if mountPassword == "" || err != nil  {
+		err = saveMountPassword()
+		if err != nil {
+			return errors.Wrap(err, "save password to wincred")
+		}
+	}
+
+	mountCmd := fmt.Sprintf("sudo mkdir -p %s && sudo mount.cifs //%s/%s %s -o username=%s,password=%s,domain=%s",m.VMDestinationPath, hostname, m.HostShareName, m.VMDestinationPath, username, mountPassword , domain)
+	rr, err := r.RunCmd(exec.Command("/bin/bash", "-c",mountCmd))
+	if err != nil {
+		errMsg := fmt.Sprintf("cifs mount. %s failed: err=%s, output: %q",mountCmd, err, rr.Output())
+		return errors.Wrap(err, errMsg)
+	}
+	glog.Infof("%s output: %q", mountCmd, rr.Output())
+	return nil
+}
+
+func (m *Cifs) Unmount(r mountRunner) error {
+	// Unmount the minikube destination path
+	cmd := UmountCmd(m.VMDestinationPath)
+	glog.Infof("Will run: %s", cmd)
+	rr, err := r.RunCmd(exec.Command("/bin/bash", "-c",cmd))
+	glog.Infof("unmount force err=%v, out=%s", err, rr.Output())
+	if err != nil {
+		return errors.Wrap(err, rr.Output())
+	}
+	return nil
+}
+
+// Save the password for mounting later on.
+func saveMountPassword () error {
+	user, err := hyperv.CurrentWindowsUser()
+	if err != nil {
+		return errors.Wrap(err,"get current user")
+	}
+	out.T(out.Notice,"In order to mount seamlessly from next time onwards, minikube requires your current password.")
+	out.T(out.Enabling,"Please type in the password for the user - [{{.username}}]",out.V{"username":user})
+	inputPassword, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+
+	if err != nil {
+		return errors.Wrap(err, "open terminal")
+	}
+
+	cred := wincred.NewGenericCredential(credentialName)
+	cred.CredentialBlob = inputPassword
+	err = cred.Write()
+	if err != nil {
+		return err
+	} else {
+		return nil
+	}
+}
+
+func getMountPassword() (string, error){
+	// Check if the Credential exists in the credential store.
+	cred, err := wincred.GetGenericCredential(credentialName)
+	if err == nil {
+		glog.Infof("Credential %v was found in the Windows Credential Store. Using that...",credentialName)
+		return string(cred.CredentialBlob), nil
+	} else {
```

Comment:
Handle and return errors first rather than using an else statement: this way the exceptional path is highlighted, and the non-exceptional one is not indented.